### PR TITLE
Add texture height blending

### PR DESCRIPTION
**Pull request by ZaPx64**
I need this feature for my game, but this PR might be valuable to the main repo.
This branch adds heightmap based texture blending. The intensity can be adjusted with the new "Height Blend Factor" parameter in the textures section of the terrain (0 = default linear blending). Recommended value is 10. Exaggerated values may lead to interesting artistic effects. I couldn't get normal blending to work.

**Blend Factor 0: (classic blending)**
blend0

**Blend Factor 10:**
blend10

### CODE REVIEW

**Comment by spimort** [comment]
Hey that looks really nice @ZaPx64 ! I'll review the PR later tonight or tomorrow.
In the mean time, could you update the Readme file in your PR to add the height blending information.

And please, make sure to take credit for it, you deserve it, write your name next to the feature 🙂

File: addons/terrabrush/Scripts/EditorResources/TextureSetResource.cs [hunk 1]
```diff
@@ -16,35 +17,51 @@ public partial class TextureSetResource : Resource {
...
             }
 
             _albedoTexture = value;
         }
     }
+
+    private Texture2D FindTexture(string fileHint, string directory, string[] directoryFiles) {
+        var files = directoryFiles.Where(file => file.Contains(fileHint, System.StringComparison.InvariantCultureIgnoreCase) && !file.EndsWith(".import"));
+        if (files.Count() == 1)
+        {
```
Comment (spimort) [inline addons/terrabrush/Scripts/EditorResources/TextureSetResource.cs:56]:
```suggestion
        if (files.Count() == 1) {
```

File: addons/terrabrush/Scripts/EditorResources/TextureSetResource.cs [hunk 2]
```diff
@@ -16,35 +17,51 @@ public partial class TextureSetResource : Resource {
...
-                }
+                RoughnessTexture ??= FindTexture(RoughnessFilesHint, directory, directoryFiles);
+                HeightTexture ??= FindTexture(HeightFilesHint, directory, directoryFiles);
             }
 
             _albedoTexture = value;
         }
     }
+
+    private Texture2D FindTexture(string fileHint, string directory, string[] directoryFiles) {
```
Comment (spimort) [inline addons/terrabrush/Scripts/EditorResources/TextureSetResource.cs:53]: Can you move this function after the properties 🙂

**Review by spimort — COMMENTED** [review]
Awesome job @ZaPx64 ! I wrote few suggestions just to fit with the code structure, nothing major

File: addons/terrabrush/Scripts/EditorResources/TextureSetResource.cs (hunk 2 above)
Comment (ZaPx64) [reply addons/terrabrush/Scripts/EditorResources/TextureSetResource.cs:52]: Done

**Pull request merged**